agent/consul/state/index_connect.go

```diff
@@ -23,7 +23,7 @@ func (idx *IndexConnectService) FromObject(obj interface{}) (bool, []byte, error
 	switch {
 	case sn.ServiceKind == structs.ServiceKindConnectProxy:
 		// For proxies, this service supports Connect for the destination
-		result = []byte(strings.ToLower(sn.ServiceProxyDestination))
+		result = []byte(strings.ToLower(sn.ServiceProxy.DestinationServiceName))
```

Comment:
Is it possible for `sn.ServiceProxy` to be `nil` here?

Reply:
Nope see above - it's not a pointer.